iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -676,6 +688,15 @@ public void executeOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     }
   }
 
+  private static ExecutorService getDeleteExecutorService(String completeName, int numThreads) {
+    AtomicInteger deleteThreadsIndex = new AtomicInteger(0);
+    return Executors.newFixedThreadPool(numThreads, runnable -> {
```

Comment:
Iceberg API may not take care of TP lifecycle. Do you need to take care of shutting down TP in finally block after expire snapshots? Otherwise it will end up creating too may TP depending on number of executions.

Reply:
Thanx. Yes, Iceberg ain't closing it. Have addressed the comment